gobblin-utility/src/main/java/gobblin/util/AvroFlattener.java

```diff
@@ -0,0 +1,357 @@
+
+    return flattenedSchema;
+  }
+
+  /***
+   * Flatten Record schema
+   * @param schema Record Schema to flatten
+   * @param shouldPopulateLineage If lineage information should be tagged in the field, this is true when we are
+   *                              un-nesting fields
+   * @return Flattened Record Schema
+   */
+  private static Schema flattenRecord(Schema schema, boolean shouldPopulateLineage) {
+    Preconditions.checkNotNull(schema);
+    Preconditions.checkArgument(Schema.Type.RECORD.equals(schema.getType()));
+
+    Schema flattenedSchema;
+
+    List<Schema.Field> flattenedFields = new ArrayList<>();
+    if (null != schema.getFields() && schema.getFields().size() > 0) {
+      for (Schema.Field oldField : schema.getFields()) {
+        LinkedList<String> lineage = new LinkedList<>();
+        List<Schema.Field> newFields = flattenField(oldField, lineage, shouldPopulateLineage, false);
+        if (null != newFields && newFields.size() > 0) {
+          flattenedFields.addAll(newFields);
+        }
+      }
+    }
+
+    flattenedSchema = Schema.createRecord(schema.getName(), schema.getDoc(), schema.getNamespace(),
+        schema.isError());
+    flattenedSchema.setFields(flattenedFields);
+
+    return flattenedSchema;
+  }
+
+  /***
+   * Flatten Union Schema
+   * @param schema Union Schema to flatten
+   * @param shouldPopulateLineage If lineage information should be tagged in the field, this is true when we are
+   *                              un-nesting fields
+   * @return Flattened Union Schema
+   */
+  private static Schema flattenUnion(Schema schema, boolean shouldPopulateLineage) {
+    Preconditions.checkNotNull(schema);
+    Preconditions.checkArgument(Schema.Type.UNION.equals(schema.getType()));
+
+    Schema flattenedSchema;
+
+    List<Schema> flattenedUnionMembers = new ArrayList<>();
+    if (null != schema.getTypes() && schema.getTypes().size() > 0) {
+      for (Schema oldUnionMember : schema.getTypes()) {
+        // Its member might still recursively contain records
+        flattenedUnionMembers.add(flatten(oldUnionMember, shouldPopulateLineage));
+      }
+    }
+    flattenedSchema = Schema.createUnion(flattenedUnionMembers);
+
+    return flattenedSchema;
+  }
+
+  /***
+   * Flatten Record field, and compute a list of flattened fields
+   *
+   * Note: Lineage represents the source from which flattening has been done. For. eg. If the original schema is:
+   * {
+   *    "type" : "record",
+   *    "name" : "parentRecordName",
+   *    "fields" : [ {
+   *      "name" : "parentFieldRecord",
+   *      "type" : {
+   *        "type" : "record",
+   *        "name" : "nestedRecordName",
+   *        "fields" : [ {
+   *            "name" : "nestedFieldString",
+   *            "type" : "string"
+   *          }, {
+   *            "name" : "nestedFieldInt",
+   *            "type" : "int"
+   *          } ]
+   *       }
+   *     }]
+   * }
+   * The expected output schema is:
```

Comment:
Yes, thats expected. Earlier we were adding that to output but after a discussion we removed it.